### PR TITLE
selfhost/typechecker: Unify `UNKNOWN_TYPE_ID`

### PR DESCRIPTION
**Pull request by cybergsus**
Now you can get UNKNOWN_TYPE_ID from a single source of truth,
`unknown_type_id()`.

### CODE REVIEW

**Comment by jntrnr** [comment]
You may need to give this a return type as return type inference is incomplete

File: selfhost/typechecker.jakt [hunk 1]
```diff
@@ -771,6 +771,7 @@ struct CheckedCall {
     callee_throws: bool
 }
 
+function unknown_type_id() => TypeId(module: ModuleId(id: 0), id: 0)
```
Comment (TophEvich) [inline selfhost/typechecker.jakt:774]: This needs a return type (atm)
```suggestion
function unknown_type_id() -> TypeId => TypeId(module: ModuleId(id: 0), id: 0)
```

**Comment by TophEvich** [comment]
@cybergsus I've been meaning to do the same.
Just a nitpicky question, why did you leave in the UNKNOWN_TYPE_ID declarations, and not just use it in place?

**Comment by cybergsus** [comment]
I don't think a refactor of those uses is needed, it's perfectly fine to save it in a variable

**Comment by cybergsus** [comment]
But I can inline the uses, no problem ;)

**Pull request merged**